projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.html

```diff
@@ -8,33 +8,39 @@
       <div class="ms-font-m back-button-text">{{ADD_REMOVE_FIELDS | async}}</div>
   </div>
   <div class="back-button-padding"></div>
-  <div class="field-picker-container" [ngClass]="{'has-changes': areCheckboxChanges}">
-      <div class="search">
-          <input [value]="fieldFilterText" [formControl]="filterFieldsControl" placeholder="{{SEARCH_FIELDS | async}}"
-              class="searchBox">
-          <div class="icon icon-search"></div>
-          <div *ngIf="fieldFilterText && fieldFilterText.length > 0" (click)="onClearFields()"
-              class="icon icon-clear">
+  <div class="add-remove-content">
+    <lf-loader-component *ngIf="isLoading" class="lf-adhoc-loader"></lf-loader-component>
+    <div *ngIf="shouldShowErrorMessage" class="lf-adhoc-loader error-entries-text">{{ AN_ERROR_OCCURED | async }}</div>
+    <div class="field-picker-container" *ngIf="!isLoading && !shouldShowErrorMessage" [ngClass]="{'has-changes': areCheckboxChanges}">
+        <div class="search">
+            <input [value]="fieldFilterText" [formControl]="filterFieldsControl" placeholder="{{SEARCH_FIELDS | async}}"
+                class="searchBox">
+            <div class="icon icon-search"></div>
+            <div *ngIf="fieldFilterText && fieldFilterText.length > 0" (click)="onClearFields()"
+                class="icon icon-clear">
+            </div>
+        </div>
+        <cdk-virtual-scroll-viewport itemSize="25" class="pick-field">
+          <div *cdkVirtualFor="let field of displayFieldInfos, let i = index">
```

Comment:
did you test the case where there's no fields with the filter?
The cdk-virtual-scroll-viewport will exist, but won't mess with the message?

Reply:
yeah it was messing with the message but I changed it. good catch

Reply:
the solution was to add a class to the no fields message, not hide the overlay?

Reply:
This class makes the no field message takes the full space of the parent container. I will change the name of the class and also hide the overlay